actions/post_actions.ts

```diff
@@ -76,7 +76,7 @@ export function flagPost(postId: string) {
         const rhsState = getRhsState(state);
 
         if (rhsState === RHSStates.FLAG) {
-            addPostToSearchResults(postId, state, dispatch);
+            dispatch(addPostToSearchResults(postId) as Action);
```

Comment:
Why do we need to typecast ?

Reply:
without typecasting, typescript is inferring its type as Thunk
***Type '(dispatch: DispatchFunc, getState: GetStateFunc) => void' is not assignable to type 'Thunk'.***

Reply:
yes because the action is not typed correctly